### PR TITLE
Bug 1819132: Convert ansible module to openstackcli

### PR DESCRIPTION
**Pull request by MaysaMacedo**
This commit ensures we rely on the openstackcli
instead of ansible modules for UPI.

### CODE REVIEW

File: upi/openstack/04_control-plane.yaml [hunk 1]
```diff
@@ -24,18 +24,23 @@
...
   - name: 'Create the Control Plane trunks'
-    os_trunk:
-      name: "{{ os_cp_trunk_name }}-{{ item.0 }}"
-      port_id: "{{ item.1.id }}"
+    command:
+      cmd: "openstack network trunk create --parent-port {{ item.1.id }} {{ os_cp_trunk_name }}-{{ item.0 }}"
     with_indexed_items: "{{ ports.results }}"
-    when: os_networking_type == "Kuryr"
+    when:
+      - os_networking_type == "Kuryr"
```
Comment (pierreprinetti) [inline upi/openstack/04_control-plane.yaml]: yamllint complains here:
```plain
./upi/openstack/04_control-plane.yaml
  42:7      error    wrong indentation: expected 4 but found 6  (indentation)
```

They want zero indentation for arrays
```yaml
    when:
    - os_networking_type == "Kuryr"
    - "'master' not in control_plane_trunks.stdout"
```

**Comment by pierreprinetti** [comment]
@MaysaMacedo 
Do you plan on removing your (amazing... :cry: ) modules and the `/library` folder in the same PR?

**Comment by MaysaMacedo** [comment]
> @MaysaMacedo
> Do you plan on removing your (amazing... cry ) modules and the `/library` folder in the same PR?

Yes. Note that this is PR is still missing a few things:
- Bugzilla for backport
- Convert ansible modules needed for destroy cluster to openstackCLI

File: upi/openstack/down-02_network.yaml [hunk 2]
```diff
@@ -9,29 +10,43 @@
...
+    with_indexed_items: "{{ ports.stdout_lines }}"
 
   - name: 'Remove the cluster router'
     os_router:
       name: "{{ os_router }}"
       state: absent
 
+  - name: 'List cluster networks'
+    command:
+      cmd: "openstack network list --tags {{ cluster_id_tag }} -f value -c Name"
```
Comment (pierreprinetti) [inline upi/openstack/down-02_network.yaml]: nit: as multiple networks can have the same name, you might want to `-c ID` instead to avoid issues

Comment (pierreprinetti) [reply upi/openstack/down-02_network.yaml]: oh no, forget that, Ansible will never want to deal with unpredictable IDs :(

**Comment by MaysaMacedo** [comment]
/hold cancel
Tested also on osp16.

**Comment by MaysaMacedo** [comment]
/hold
Try one improvement before merging

File: upi/openstack/down-06_load-balancers.yaml [hunk 3]
```diff
@@ -9,11 +9,68 @@
...
+      headers:
+        X-Auth-Token: "{{ cloud.ansible_facts.auth_token }}"
+        Content-Type: 'application/json'
+      url: "{{ cloud.ansible_facts.service_catalog | selectattr('name', 'match', 'octavia') | first | json_query('endpoints') | selectattr('interface', 'match', 'public') | first | json_query('url') }}/"
+    register: octavia_versions
+    when: os_networking_type == "Kuryr"
+
+  - set_fact:
+      versions: "{{ octavia_versions.json.versions | selectattr('id', 'match', 'v2.5') | map(attribute='id') | list }}"
+      when: os_networking_type == "Kuryr"
```
Comment (wjiangjay) [inline upi/openstack/down-06_load-balancers.yaml:29]: bad indent here

**Pull request merged**